src/poetry/core/version/pep440/segments.py

```diff
@@ -74,6 +73,7 @@ def to_string(self) -> str:
         return self.text
 
     def next_major(self) -> Release:
+        assert isinstance(self.extra, tuple)
```

Comment:
More out of curiosity, any paritcular reason why an assert is preferred in these cases over a cast?

Reply:
I probably try to go with assert where I think there's any chance that I might be wrong, and cast only when I'm really 100% sure.  Because if I cast and am wrong, then things will break in possibly confusing ways, whereas if I assert and am wrong then things will break in a clearer way.

But I don't promise that I'm 100% consistent about this and, if this were more than curiosity, I'd likely be willing to change some or more of the choices that I've made.